### PR TITLE
Add public endpoints/paths that won't check for authentication

### PR DESCRIPTION
**Pull request by smaspons**
Please let me know if there's something wrong. I tested it and it works for me

### CODE REVIEW

**Comment by manisandro** [comment]
Thanks, could you also update the mapViewerConfig schema definition with the new setting?

**Comment by smaspons** [comment]
Yes, I forgot it

File: schemas/qwc-map-viewer.json [hunk 1]
```diff
@@ -117,6 +117,13 @@
           "items": {
             "type": "string"
           }
+        },
+        "public_paths": {
+          "description": "List of paths that will be accessible without a JWT identity",
```
Comment (manisandro) [inline schemas/qwc-map-viewer.json:122]: "Optional: List of paths that will be accessible without a JWT identity, if `AUTH_REQUIRED` is set."

**Comment by manisandro** [comment]
Thanks

File: server.py [hunk 2]
```diff
@@ -60,6 +61,17 @@ def auth_path_prefix():
...
+    if request.endpoint in public_endpoints:
+        return
+
+    tenant = tenant_handler.tenant()
+    config_handler = RuntimeConfig("mapViewer", app.logger)
+    config = config_handler.tenant_config(tenant)
+    public_paths = config.get("public_paths", [])
+    if request.path in public_paths:
+        return
+
```
Comment (tpo) [inline server.py:74]: @smaspons @manisandro - as far as I understand, the code adds two new URLs, `.../healthz` and `.../ready` that get served by the service without authentication.

That means that the map-viewer becomes discoverable without authentication, which is an information leak security-wise.
 
Additionaly it's now possible to add other patchs that would get the same treatment.

Wouldn't it be better to make `.../healthz` and `.../ready` optional as well and let the user add them to the `public_paths` if she chooses to do so?

**Pull request merged**